js_modules/dagster-ui/packages/ui-core/src/assets/AssetsGroupsGlobalGraphRoot.tsx

```diff
@@ -31,7 +31,7 @@ export const AssetsGroupsGlobalGraphRoot = () => {
   const history = useHistory();
 
   const [filters, setFilters] = useQueryPersistedState<{groups: AssetGroupSelector[]}>({
-    encode: ({groups}) => ({groups: JSON.stringify(groups)}),
+    encode: ({groups}) => ({groups: groups.length ? JSON.stringify(groups) : undefined}),
```

Comment:
This stops the empty `?groups=` from being appended to the global graph URLs all the time